hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/NamenodeFGL.md

```diff
@@ -0,0 +1,210 @@
+| Involving only blocks                  | `reportBadBlocks`, `updateBlockForPipeline`, `updatePipeline`                                                                                                                                                                                                                                                                                                                                                                                                             |
+| Involving only DNs                     | `registerDatanode`, `setBalancerBandwidth`, `sendHeartbeat`                                                                                                                                                                                                                                                                                                                                                                                                               |
+| Involving both namespace tree & blocks | `getBlockLocation`, `create` (with overwrite), `append`, `setReplication`, `abandonBlock`, `addBlock`, `getAdditionalDatanode`, `complete`, `concat`, `truncate`, `delete`, `getListing` (with locations), `getFileInfo` (with locations), `recoverLease`, `listCorruptFileBlocks`, `fsync`, `commitBlockSynchronization`, `RedundancyMonitor`, `processMisReplicatedBlocks`                                                                                              |
+| Involving both DNs & blocks            | `getBlocks`, `errorReport`                                                                                                                                                                                                                                                                                                                                                                                                                                                |
+| Involving namespace tree, DNs & blocks | `blockReport`, `blockReceivedAndDeleted`, `HeartbeatManager`, `Decommission`                                                                                                                                                                                                                                                                                                                                                                                              |
```

Comment:
@kokonguyen191 @ZanderXu Sir, i did not see any namespace tree modification in the scope of writeLock()  
 in method HeartbeatManager#heartbeatCheck. Please correct me if i mistook it.